controller/free_Board_Comments.controller.js

```diff
@@ -0,0 +1,120 @@
+const freeBoardCommentsRepository = require("../repository/freeBoardCommentsRepository");
+const authorization = require("../middleware/authorization");
+const { ACCESS_PRIVATE_KEY } = process.env;
+
+async function insertFreeBoardComments(req, res) {
+    try {
+        const decodedToken = authorization.verifyToken(
+            req.headers["authorization"],
+            ACCESS_PRIVATE_KEY
+        );
+
+        if (decodedToken) {
+            const { postId } = req.params;
+            const { body } = req;
+            const row = await freeBoardCommentsRepository.insertFreeBoardComments(decodedToken, postId, body);
+
+            if (row.affectedRows === 1) {
+                return res.status(200).json({ message: "댓글 작성 완료" });
+            }
+        }
+    } catch(error) {
+        console.log(error);
+        return res.status(500).json({ message: "댓글 작성 권한이 없습니다." });
+    }
+}
+
+async function putFreeBoardComments(req, res) {
+    try {
+        const decodedToken = authorization.verifyToken(
+            req.headers["authorization"],
+            ACCESS_PRIVATE_KEY
+        );
+
+        const { postId, commentId } = req.params;
+        const selectRow = await freeBoardCommentsRepository.selectFreeBoardComments(commentId, postId);
+
+        if (decodedToken.id === selectRow.user_id) {
+            const { body } = req;
+            const updateRow = await freeBoardCommentsRepository.updateFreeBoardComments(decodedToken, postId, commentId, body);
+
+            if (updateRow.affectedRows === 1) {
+                return res.status(200).json({ message: "댓글 수정 완료" });
+            }
+        }
```

Comment:
만약 A유저가 본인의 토큰으로 B의 게시글을 수정하려고 한다면  
오류는 발생하지 않고 return도 없고 응답도 없이 함수가 종료됩니다.
이럴때 서버는 응답을 보내지 않은상태로 연결을 유지 합니다.
```suggestion
        if (decodedToken.id === selectRow.user_id) {
            const { body } = req;
            const updateRow = await freeBoardCommentsRepository.updateFreeBoardComments(decodedToken, postId, commentId, body);

            if (updateRow.affectedRows === 1) {
                return res.status(200).json({ message: "댓글 수정 완료" });
            }
        }else{
            return res.status(403).json({ message: "수정권한이 없습니다" });
        }
```